tfx_addons/predictions_to_bigquery/README.md

```diff
@@ -0,0 +1,127 @@
+# Prediction results to BigQuery component
+
+[![Python](https://img.shields.io/pypi/pyversions/tfx.svg?style=plastic)](https://github.com/tensorflow/tfx)
+[![TensorFlow](https://img.shields.io/badge/TFX-orange)](https://www.tensorflow.org/tfx)
+
+## Project Description
+
+This component exports prediction results from BulkInferrer to a BigQuery
+table.
+The BigQuery table schema can be generated through one of the following sources:
+1. From SchemaGen component output
+2. From Transform component output
+3. From BulkInferrer component output (i.e. prediction results)
+
+If both SchemaGen and Transform outputs are passed to the component,
+the SchemaGen output will take priority. It would be best to use SchemaGen
+for generating the BigQuery schema.
+
+If the Transform output channel is passed to the component, without the
+SchemaGen output, the BigQuery schema will be derived from the pre-transform
+metadata schema generated by Transform. Note that the metadata schema may
+include a label key, which may not be present in the BulkInferrer prediction
+results. Therefore, this option may not work for unlabeled data.
+
+If neither the SchemaGen nor Transform outputs are passed to the component,
+the BigQuery schema will be parsed from the BulkInferrer prediction results
+itself, which contains tf.Example protos.
+
+Prediction string labels from the BulkInferrer output may be derived by passing a 'vocab_label_file' execution parameter to the component. This will only work
+if the Transform component output is passed and if it the `vocab_label_file`
+is present.
+
+## Project Use-Case(s)
+
+The main use case for this components is to enable export of model prediction
+results into a BigQuery for further data analysis. The exported table will
+contain the model predictions and their corresponding inputs. If the input
+data is labeled, this would allow users to compare labels and corresponding predictions.
+
+## Project Implementation
+
+PredictionsToBigQuery component uses Beam to process the prediction results
+from BulkInferrer and export it to a BigQuery table.
+
+The BigQuery table name is passed as a parameter by the user, however the user
+can also choose to have the component append a timestamp at the end of the table name.
+
+The output component is the fully qualified BigQuery table name where the inference results are stored, and this can be accessed through the `bigquery_export` key. The same table name is also stored as a custom property
+of the `bigquery_export` artifact.
+
+### Usage example
+
+```python
+
+from tfx import v1 as tfx
+import tfx_addons as tfxa
+
+...
+
+predictions_to_bigquery = tfxa.predictions_to_bigquery.PredictionsToBigQuery(
+    schema=schema_gen.outputs['schema']
+    transform_graph=transform.outputs['transform_graph'],
+    bq_table_name='my_bigquery_table',
+    gcs_temp_dir='gs://bucket/temp-dir',
```

Comment:
Should this just use the temp_dir from beam_pipeline_args instead?

Reply:
Yes, I just realized that the `WriteToBigQuery` transform used by the Beam pipeline defaults to `temp_location` if `custom_gcs_temp_location` is not specified, and `temp_location` should be an argument supported by `beam_pipeline_args`. Although I'm not quite sure if `temp_location` needs to be explicitly set or if Beam will create one by default otherwise.
I think this would be a more involved change as the Vertex integration test needs a custom container having the pred2bq component in Artifact Registry to run. Perhaps we have it as a separate PR instead?